### PR TITLE
Add widget for COVID vaccine updates

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR adds a new Flipper widget that prompts the user to sign up for COVID vaccine updates.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/17102#issuecomment-741114051

## Testing done
I added `<div data-widget-type="va-covid-vaccine-updates-cta"></div>` into a local page in my vagov-content repo. Then I navigated to that page and toggled the feature in my local API `http://localhost:3000/flipper/features/covid_vaccine_registration_frontend_cta`.

### Off state
image

### On state
image

## Acceptance criteria
- [ ] Widget is ready

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/static-pages/covid-vaccine-updates-cta/createCovidVaccineUpdatesWidget.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+export default function createCovidVaccineUpdatesWidget(store, widgetType) {
+  const root = document.querySelector(`[data-widget-type="${widgetType}"]`);
+  if (root) {
+    import(/* webpackChunkName: "CaregiverContentToggle" */
```
Comment (U-DON) [inline src/applications/static-pages/covid-vaccine-updates-cta/createCovidVaccineUpdatesWidget.js:7]: Is this the correct chunk name?

Comment (ncksllvn) [reply src/applications/static-pages/covid-vaccine-updates-cta/createCovidVaccineUpdatesWidget.js:7]: Lol - no it is not! One sec

File: src/applications/static-pages/covid-vaccine-updates-cta/widget.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import OnState from './On';
+import OffState from './Off';
+
+function CovidVaccineUpdatesCTA({ showLinkToOnlineForm }) {
+  if (showLinkToOnlineForm) {
+    return <OnState />;
+  }
+
+  if (showLinkToOnlineForm === false) {
```
Comment (kelsonic) [inline src/applications/static-pages/covid-vaccine-updates-cta/widget.jsx:15]: Love how you wrote this, super easy to read 💯

**Review by U-DON — APPROVED** [review]

File: src/applications/static-pages/covid-vaccine-updates-cta/createCovidVaccineUpdatesWidget.js (hunk 1 above)
Comment (kelsonic) [inline src/applications/static-pages/covid-vaccine-updates-cta/createCovidVaccineUpdatesWidget.js]: Confirming here that the webpack chunk name is correct

Comment (ncksllvn) [reply src/applications/static-pages/covid-vaccine-updates-cta/createCovidVaccineUpdatesWidget.js]: now it is!

Comment (ncksllvn) [reply src/applications/static-pages/covid-vaccine-updates-cta/createCovidVaccineUpdatesWidget.js]: just pushed a rename!

File: src/applications/static-pages/covid-vaccine-updates-cta/On.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          <p>
+            When you sign up, we'll also ask about your interest in getting a
+            vaccine when one is available to you. By sharing your interest, you
+            can help us better prepare as we work to offer vaccines to more
+            Veterans.
+          </p>
+          <p>
+            <strong>Note:</strong> You don’t need to sign up to get a vaccine.
+          </p>
+          <a href={covidVaccineFormUrl} className="usa-button">
```
Comment (kelsonic) [inline src/applications/static-pages/covid-vaccine-updates-cta/On.jsx:27]: Could you check the spacing for this element? I remember needing to wrap a `p` tag around these floating `a` tags to ensure the spacing was correct, but not sure if `usa-button` does that for you already 🙂

**Review by kelsonic — APPROVED** [review]
Looks great! Well done! 🎉 Left a few confirmation comments 🙂

File: src/applications/static-pages/covid-vaccine-updates-cta/On.jsx (hunk 3 above)
Comment (ncksllvn) [reply src/applications/static-pages/covid-vaccine-updates-cta/On.jsx:27]: It looks Like it does add the margin but yeah good call -

image

**Review by stephenBarrs — APPROVED** [review]

File: src/applications/static-pages/covid-vaccine-updates-cta/createCovidVaccineUpdatesWidget.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+export default function createCovidVaccineUpdatesWidget(store, widgetType) {
+  const root = document.querySelector(`[data-widget-type="${widgetType}"]`);
+  if (root) {
+    import(/* webpackChunkName: "CovidVaccineUpdatesCTA" */
```
Comment (erikphansen) [inline src/applications/static-pages/covid-vaccine-updates-cta/createCovidVaccineUpdatesWidget.js:7]: @ncksllvn @kelsonic I am trying to figure out where this chunk name comes from.

**Pull request merged**